frontend/src/hooks/useActions.js

```diff
@@ -114,12 +123,105 @@ const useActions = (registerHotkeys = false) => {
       handler: redo
     },
     COPY: {
-      hotkey: { key: 'c', meta: true }
-      // handler: () => console.log('Copy'),
+      hotkey: { key: 'c', meta: true },
+      handler: () => {
+        const selectedStates = selectedStatesIds.map((stateId) => {
+          return project.states.find((state) => {
+            return state.id === stateId
+          })
+        })
+        const selectedComments = selectedCommentsIds.map((commentId) => {
+          return project.comments.find((comment) => {
+            return comment.id === commentId
+          })
+        })
+        const selectedTransitions = selectedTransitionsIds.map((transitionId) => {
+          return project.transitions.find((transition) => {
+            return transition.id === transitionId
+          })
+        })
+        const isInitialSelected = selectedStatesIds.includes(project.initialState)
+        // This will use the CopyData type defined in ProjectTypes
+        const copyData = {
+          states: selectedStates,
+          comments: selectedComments,
+          transitions: selectedTransitions,
+          projectSource: project._id,
+          projectType: project.projectType,
+          initialStateId: isInitialSelected ? project.initialState : null
+        }
+        sessionStorage.setItem(COPY_DATA_KEY, JSON.stringify(copyData))
```

Comment:
Think `localStorage` might be more appropriate?

One problem I found with `sessionStorage` when testing the branch was that [sessionStorage is per tab and doesn't really sync well if you have two tabs open](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API#web_storage_concepts_and_usage). Imo the times I'd want to be copying/pasting between projects I'd have multiple tabs open so would be good to have it work in those cases

Reply:
My bad, I got ahead of myself and forgot to test again between tabs